### ISSUE TITLE
User scripts conflict with system scripts

### ISSUE THREAD

**Issue by lucaderi** [ntop/ntopng#2928]
After the migration of the scripts (flow and alert scripts) user scripts must be into the same directory used by the package. This not wrong as they must stay on a separate directory that
- does not conflict with the package
- does not live under /etc/ntopng
- is independent from the packaging directories

It is requested to design this feature, update the packages for all the platforms if necessary (probably not), and make sure that ntopng will also poll this new path.

**Comment by simonemainardi** [comment]
Candidates are:

`/usr/lib/ntopng`: This directory contains program libraries. Libraries are collections of frequently used program routines. (https://www.tldp.org/LDP/Linux-Filesystem-Hierarchy/html/usr.html)
￼
Also re-using `/usr/share` is consistent (This directory contains 'shareable', architecture-independent files), where we could create a subdirectory `/ntopng/scripts/callbacks/custom/`.

As it can be seen from the following snapshots, any custom scripts placed won't be deleted even when option --purge is used.

```
simone@devel:~$ ls -lha /usr/share/ntopng/scripts/callbacks/custom/
total 8.0K
drwxr-xr-x 2 root root 4.0K Oct  7 12:43 .
drwxrwxr-x 5 root root 4.0K Oct  7 12:43 ..
-rw-r--r-- 1 root root    0 Oct  7 12:43 cane
simone@devel:~$ sudo apt-get remove ntopng
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following package was automatically installed and is no longer required:
  icinga2-doc
Use 'sudo apt autoremove' to remove it.
The following packages will be REMOVED:
  ntopng
0 upgraded, 0 newly installed, 1 to remove and 25 not upgraded.
After this operation, 43.9 MB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 343588 files and directories currently installed.)
Removing ntopng (3.9.191007-7737) ...
Stopping ntopng services...
Processing triggers for man-db (2.7.5-1) ...
simone@devel:~$ ls -lha /usr/share/ntopng/scripts/callbacks/custom/
total 8.0K
drwxr-xr-x 2 root root 4.0K Oct  7 12:43 .
drwxrwxr-x 3 root root 4.0K Oct  7 12:43 ..
-rw-r--r-- 1 root root    0 Oct  7 12:43 cane
simone@devel:~$ sudo apt-get --purge remove ntopng
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following package was automatically installed and is no longer required:
  icinga2-doc
Use 'sudo apt autoremove' to remove it.
The following packages will be REMOVED:
  ntopng*
0 upgraded, 0 newly installed, 1 to remove and 25 not upgraded.
After this operation, 43.9 MB disk space will be freed.
Do you want to continue? [Y/n] y
(Reading database ... 343588 files and directories currently installed.)
Removing ntopng (3.9.191007-7737) ...
Stopping ntopng services...
Removed symlink /etc/systemd/system/multi-user.target.wants/ntopng.service.
Purging configuration files for ntopng (3.9.191007-7737) ...
dpkg: warning: while removing ntopng, directory '/usr/share/ntopng/scripts/callbacks' not empty so not removed

**Comment by lucaderi** [comment]
Done this ticket, #2953 can be implemented.

**Comment by emanuele-f** [comment]
Implemented in a3432e00e89f461492b033940bbb46f09bab2fb7 . The datadir subdir `/var/lib/ntopng/plugins` is used for now
